src/hdmf/build/classgenerator.py

```diff
@@ -0,0 +1,336 @@
+                    kwargs.update(name=name)
+                pargs, pkwargs = fmt_docval_args(base.__init__, kwargs)
+                base.__init__(self, *pargs, **pkwargs)  # special case: need to pass self to __init__
+
+                for f in new_args:
+                    arg_val = kwargs.get(f, None)
+                    setattr(self, f, arg_val)
+
+            classdict.update(__init__=__init__)
+
+        return classdict
+
+    @classmethod
+    def update_cls_args(cls, classdict, bases, not_inherited_fields, spec, type_map):
+        """Add __fields__ and __init__ to the classdict
+        :param classdict: The dict to update with __clsconf__ if applicable
+        :param bases: The list of base classes to update if applicable
+        :param not_inherited_fields: Dict of additional fields that are not in the base class
+        :param spec: The spec for the container class to generate
+        :param type_map: TypeMap to use when determining type for a spec
+        """
+        parent_cls = bases[0]
+        d = cls._build_fields(parent_cls, not_inherited_fields, type_map)
+        classdict.update(d)
+        d = cls._build_init(parent_cls, not_inherited_fields, type_map, spec.name, spec.default_name)
+        classdict.update(d)
+        # TODO add test that fields is built correctly
+
+
+class MCIClassGenerator(CustomClassGenerator):
+
+    @classmethod
+    def update_cls_args(cls, classdict, bases, not_inherited_fields, spec, type_map):
+        """Update the given class dict and base classes if there is a field spec with quantity * or +
+        :param classdict: The dict to update with __clsconf__ if applicable
+        :param bases: The list of base classes to update if applicable
+        :param not_inherited_fields: Dict of additional fields that are not in the base class
+        :param spec: The spec for the container class to generate
+        :param type_map: The type map to use
+        """
+        # NOTE: spec is not used here
+
+        # create MultiContainerInterface __clsconf__
+        clsconf = list()
+        for f, field_spec in not_inherited_fields.items():
+            if getattr(field_spec, 'quantity', None) in (ZERO_OR_MANY, ONE_OR_MANY):
+                clsconf.append(dict(
+                    attr=f,
+                    type=cls._get_type(field_spec, type_map),  # <-- from ClassGenerator
+                    add='add_{}'.format(f),
+                    get='get_{}'.format(f),
+                    create='create_{}'.format(f)
+                ))
+                # remove the attribute from __fields__ -- needed so that MCI initializes the attribute to LabelledDict
+                classdict['__fields__'] = tuple(filter(lambda x: x['name'] != f, classdict['__fields__']))
+
+        if len(clsconf):
+            classdict.update(__clsconf__=clsconf)
+
+            # add MultiContainerInterface to the type hierarchy if not already there
+            if not isinstance(bases[0], MultiContainerInterface):
```

Comment:
when does this happen?

Reply:
Good point. I can't think of a scenario.